inc/rain.tpl.class.php

```diff
@@ -478,7 +478,7 @@ protected function compileCode( $parsed_code ){
 				$value = "\$value$loop_level";           // value
 
 				//loop code
-				$compiled_code .=  "<?php $counter=-1; if( isset($var) && is_array($var) && sizeof($var) ) foreach( $var as $key => $value ){ $counter++; ?>";
+				$compiled_code .=  "<?php $counter=-1; if( $var !== null && is_array($var) && sizeof($var) ) foreach( $var as $key => $value ){ $counter++; ?>";
```

Comment:
@Phyks instead of ```$var !== null``` we can use ```is_null($var)```, that works with both undefined variables and return of function calls.